pkg/v1/providers/ytt/02_addons/csi/csi_addon_data.lib.yaml

```diff
@@ -1,6 +1,26 @@
 #@ load("@ytt:data", "data")
+#@ load("@ytt:regexp", "regexp")
+#@ load("@ytt:assert", "assert")
 #@ load("/lib/helpers.star", "get_bom_data_for_tkr_name", "get_image_repo_for_component", "get_no_proxy")
 
+
+#! Set topologyFeatureFlag to be true for K8S 1.22 and newer
+#@ def enableTopologyFeature():
+#@ if regexp.match("^v(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$", data.values.KUBERNETES_VERSION) != True:
```

Comment:
Suggestion: Maybe let's use TKR version instead ? Although TKR version is usually aligned with K8S version, semantically whether passing in `true` of this flag is based on CSI version, and CSI version is decided by TKR. 

Using K8S version for comparison works, but people might wonder if this flag is to fix some compatibility issue between CSI and K8S, however they are actually unrelated.

Reply:
It seems that BOM file doesn't have a valid TKR version available.

e.g.
```
apiVersion: run.tanzu.vmware.com/v1alpha2
default:
  k8sVersion: v1.21.5+vmware.1-tkg.1-20211029
release:
  version: v1.5.0-20211029
```
We can only infer the TKR major, minor and patch version by K8S version.

Reply:
I was looking at TKG BOM, should look at TKR BOM instead. Changed the reference to TKR version.